### DISCUSSION TITLE
How do I get Affected Rows

### DISCUSSION THREAD

**Discussion by Honst211** [sqlalchemy/sqlalchemy#7031] [Usage Questions]
I want to get the actual number of rows mysql affects

**Comment by CaselIT** [comment]
Hi,

More information is needed for a better answer. Please try searching the documentation:
- https://docs.sqlalchemy.org/en/14/tutorial/data_update.html#tutorial-update-delete-rowcount
- https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.BaseCursorResult.rowcount

**Reply by Honst211** [reply]
I know that I need the number of affected rows. All rows are recorded here

**Reply by CaselIT** [reply]
I don't know what "number of affected rows" means , sorry.

Can you provide an example?

**Reply by Honst211** [reply]
For example, I modified 100 pieces of data and actually modified 10 pieces. What I want to obtain is 10 pieces of data
![image](https://user-images.githubusercontent.com/50348812/133887487-a2806797-d1e3-4d79-acbb-8a1a4f568fc0.png)
I need that affected rows;

**Reply by CaselIT** [reply]
I think that's just the rowcount.

Sqlalchemy has only that information, you may be able to obtain more detailed data by using the dbapi connection directly: https://docs.sqlalchemy.org/en/14/faq/connections.html#how-do-i-get-at-the-raw-dbapi-connection-when-using-an-engine
